### ISSUE TITLE
Zero delay retry loop

### ISSUE THREAD

**Issue by mideeks** [Azure/azure-sdk-for-python#485]
NetworkResourceProviderClient.get_long_running_operation_status() does not set a value for result.retry_after. When long running operations start looping and checking the status they get None for this value. Then in places like NetworkInterface.create_or_update [line 7124], this check fails:
```
if delay_in_seconds == 0:
  delay_in_seconds = 15
```

This has the nasty potential of eating up my rate limit quota and locking me out from further API calls for an hour.

**Comment by crwilcox** [comment]
Hey Mike.  Thanks for finding this.

Did you see delay_in_seconds set to None or is that an assumption?  I only ask since time.sleep complains if you pass it anything that is not a positive number.  If it were getting None at that point you would see a TypeError. Particularly line 86 to add a default value.

` self._retry_after = kwargs.get('retry_after', 0)`

I am guessing you are using the code from pypi though and not this repo?  There was a change to networkresourceprovider.py around a week ago https://github.com/Azure/azure-sdk-for-python/commit/f7f90121099a699b010fe50a76ea7bd3526d9473.

**Comment by mideeks** [comment]
I got the latest from pypi after I first saw the bug. I tried again after updating and it still exists. I'm pretty sure it was None because I manually put a print() in there to check.

Sent from Outlook<http://aka.ms/Ox5hz3>




On Thu, Oct 1, 2015 at 8:58 PM -0700, "Christopher Wilcox" <notifications@github.com<mailto:notifications@github.com>> wrote:


Hey Mike. Thanks for finding this.

Did you see delay_in_seconds set to None or is that an assumption? I only ask since time.sleep complains if you pass it anything that is not a positive number. If it were getting None at that point you would see a TypeError. Particularly line 86 to add a default value.

self._retry_after = kwargs.get('retry_after', 0)

I am guessing you are using the code from pypi though and not this repo? There was a change to networkresourceprovider.py around a week ago f7f9012<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fAzure%2fazure-sdk-for-python%2fcommit%2ff7f90121099a699b010fe50a76ea7bd3526d9473&data=01%7c01%7cmideeks%40microsoft.com%7c7f00f8e9a07e4bcbde9a08d2caddb179%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=9BQHXb8AAeY7w6tJP0Efrd%2bnAyVs%2bnATb5XxhF%2bb%2fAQ%3d>.

-
